packages/markers/src/browser/problem/problem-widget.ts

```diff
@@ -69,11 +73,22 @@ export class ProblemWidget extends TreeWidget {
         }
     }
 
-    protected onUpdateRequest(msg: Message): void {
+    protected onUpdateRequest(msg: Message) {
         if (!this.model.selectedNode && ISelectableTreeNode.is(this.model.root)) {
             this.model.selectNode(this.model.root);
         }
-        super.onUpdateRequest(msg);
+        this.setWorkspacePath().then(() => {
+            super.onUpdateRequest(msg);
+        });
+    }
+
+    protected async setWorkspacePath() {
+        if (!this.workspacePath) {
```

Comment:
We don't need local state here, we can turn this method into `getWorkspacePath` always internally calling tryRoot.